src/ClassDiscovery.php

```diff
@@ -120,9 +121,9 @@ public static function setStrategies(array $strategies)
     /**
      * Append a strategy at the end of the strategy queue.
      *
-     * @param DiscoveryStrategy $strategy
+     * @param string $strategy FQN to a DiscoveryStrategy
      */
-    public static function appendStrategy(DiscoveryStrategy $strategy)
+    public static function appendStrategy($strategy)
```

Comment:
Add a resource ONLY in an appended strategy and see if you can resolve that resource.